testcase/unittest/blockchain/blocks/v1_0/test_block_verifier.py

```diff
@@ -0,0 +1,242 @@
+                monkeypatch.setattr(block_verifier, method, MagicMock())
+
+        return block_verifier
+
+
+@pytest.mark.asyncio
+class TestVerifyVersion(_TestVerifierBase):
+    @property
+    def target_method(self):
+        return "_verify_version"
+
+    async def test_block_version_dismatched(self, block_verifier, create_block):
+        # GIVEN I have blocks
+        prev_block = create_block()
+
+        # AND its version is not matched to verifier
+        block = create_block(prev_block=prev_block, version="0.5")
+        assert block_verifier.version != block.header.version
+
+        # WHEN I verify the block
+        with pytest.raises(BlockVersionNotMatch):
+            # THEN verification failed
+            await block_verifier.verify(prev_block, block)
+
+
+@pytest.mark.asyncio
+class TestVerifyCommon(_TestVerifierBase):
+    @property
+    def target_method(self):
+        return "_verify_common"
+
+    async def test_block_timestamp_not_exist(self, block_verifier, create_block):
+        # GIVEN I have blocks
+        prev_block = create_block()
+
+        # AND its timestamp is not set
+        block = create_block(prev_block, timestamp=None)
+        assert block.header.timestamp is None
+
+        # WHEN I verify the block
+        with pytest.raises(RuntimeError, match="Block.* timestamp."):
+            # THEN verification failed
+            await block_verifier.verify(prev_block, block)
+
+    async def test_not_genesis_block_has_no_prev_hash(self, block_verifier, create_block):
+        # GIVEN I have blocks
+        prev_block = create_block()
+
+        # AND it is not genesis block
+        block = create_block(prev_block, height=1, prev_hash=None)
+        assert block.header.height != 0
+        # AND its prev_hash is not set
+        assert block.header.prev_hash is None
+
+        # WHEN I verify the block
+        with pytest.raises(RuntimeError, match="Block.* prev_hash."):
+            # THEN verification failed
+            await block_verifier.verify(prev_block, block)
+
+    async def test_block_prev_hash_not_exist(self, block_verifier, create_block):
+        # GIVEN I have blocks
+        prev_block = create_block()
+
+        # AND it is not genesis block
+        block = create_block(prev_block, height=2, prev_hash=None)
+        assert block.header.height != 0
+        # AND its prev_hash is not set
+        assert block.header.prev_hash is None
+
+        # WHEN I verify the block
+        with pytest.raises(RuntimeError, match="Block.* prev_hash."):
+            # THEN verification failed
+            await block_verifier.verify(prev_block, block)
+
+    @pytest.mark.xfail(reason="Validate reps on BlockVerifier or in Epoch?")
+    async def test_reps(self, block_verifier, create_block):
+        assert 0
+
+    @pytest.mark.xfail(reason="What should be tested on verify_prev_votes?")
+    async def test__verify_prev_votes(self, block_verifier, create_block):
+        assert 0
```

Comment:
Please complete those test case if we decided to verify about validators and prev_votes.